### DISCUSSION TITLE
PSQL Constraint for Parent Children Relationship

### DISCUSSION THREAD

**Discussion by sandbox-apps** [supabase/supabase#2359] [Questions] [answered]
Hi all @supabase users is there any way I can make a constraint to validate the integrity of a relationship?
I want the parent table to have atleast minimum of 2 children upon insert (Optionally can also specify max but im focused on the children minimum constraint)

Use Case:
Polls Table where you can create a Poll with atleast 2 options...
I can make a table with columns of:
optionA Text (required)
optionB Text (required)
optionC Text (optional)
optionD Text (optional)

This is ok, but with some limitation, as you can see you are limited to max of 4 options. Typical polling system can add more options.

How can I achieve this? I want to achieve this at db level so I can make use of supabase studio insert row functionality with validation... Thanks

**Comment by steve-chavez** [comment, marked as answer]
Hey @sandbox-apps,

**Edit**: an array was the solution here, see below.

For your case, I think you could create a trigger that enforces the constraint(does a `raise foreign_key_violation` inside).

Hard to come with a snippet without a sample schema, but there's an example [here](https://postgrest.org/en/stable/auth.html?highlight=check_role_exists#storing-users-and-passwords)(see the `check_role_exists()` trigger function).

**Reply by sandbox-apps** [reply]
Well the schema would be something link this that involves two table

```ts
class Poll {
  id: string
  text: string
}

class Option {
  pollId: string
  id: string
  text: string
}
```

But I'll try your idea thanks @steve-chavez

**Reply by steve-chavez** [reply]
@sandbox-apps Hm, instead of a table couldn't you make `options` be an array? Then enforcing the presence of two polls would be a simple check constraint. Like:

```sql
create table polls(
  id text primary key,
  txt text,
  options text[] CHECK (cardinality(options) >= 2)
);
```

**Reply by sandbox-apps** [reply]
Thanks!!! This will be okay to. I can also limit by max 10 options here :D
